app_test.go

```diff
@@ -1,5 +1,72 @@
 package buffalo
 
+import (
+	"net/http"
+	"testing"
+	"time"
+)
+
 func voidHandler(c Context) error {
 	return nil
 }
+
+func TestGracefulShutdown(t *testing.T) {
+	requestFinished := false
+
+	app := New(NewOptions())
+	app.GET("/slow", func(c Context) error {
+		time.Sleep(4 * time.Second)
+		requestFinished = true
+		return nil
+	})
+
+	go func() {
+		time.Sleep(2 * time.Second)
+		app.Stop(nil)
+	}()
+
+	go func() {
+		time.Sleep(1 * time.Second)
+		_, err := http.Get("http://localhost:17171/slow")
+		if err != nil {
```

Comment:
Can you please use the `require` assertion library, like the rest of the tests? We like to keep things consistent.

Reply:
will do -- `require` readability is much better